plasmapy/utils/pytest_helpers.py

```diff
@@ -719,3 +720,145 @@ def run_test_equivalent_calls(*test_inputs, require_same_type: bool = True):
+
+    Examples
+    --------
+    >>> from plasmapy.physics.parameters import Alfven_speed, gyrofrequency
+    >>> assert_can_handle_nparray(Alfven_speed)
+    >>> assert_can_handle_nparray(gyrofrequency, kwargs={"signed": True})
+    >>> assert_can_handle_nparray(gyrofrequency, kwargs={"signed": False})
+    """
+    def _prepare_input(param_name, param_default, insert_some_nans, insert_all_nans, kwargs):
+        """
+        Parse parameter names and set up values to input for 0d, 1d, and 2d array tests.
+        """
+        # first things first: let any passed in kwarg right through (VIP access)
+        if param_name in kwargs.keys():
+            return kwargs[param_name], kwargs[param_name], kwargs[param_name]
+
+        # else, if it's a recognized variable name, give it a reasonable unit and magnitude
+        elif param_name in ["particle", "ion_particle", "ion"]:
+            if not (param_default is inspect._empty or param_default is None):
+                return param_default, param_default, param_default
+            else:
+                return "p", "p", "p"
+        elif param_name == "particles":
+            if not (param_default is inspect._empty):
+                return param_default, param_default, param_default
+            else:
+                return ("e", "p"), ("e", "p"), ("e", "p")
+        elif param_name in ["T", "T_i", "T_e", "temperature"]:
+            unit = u.eV
+            magnitude = 1.0
+        elif param_name in ["n", "n_i", "n_e", "density"]:
+            unit = u.m ** -3
+            magnitude = 1e20
+        elif param_name == "B":
+            unit = u.G
+            magnitude = 1e3
+        elif param_name in ["V", "Vperp"]:
+            unit = u.m / u.s
+            magnitude = 1e5
+        elif param_name == "coulomb_log":
+            unit = 1.0
+            magnitude = 1e1
+        elif param_name == "characteristic_length":
+            unit = u.m
+            magnitude = 1.0
+
+        # else, last resort, if it has a default argument, go with that:
+        elif not (param_default is inspect._empty):
+            return param_default, param_default, param_default
+
+        else:
+            raise ValueError("Unrecognized function input")
+
+        # now knowing unit and magnitude, set up the 0d, 1d, and 2d arrays:
+        input_data_2d = np.reshape(np.arange(1.0, 5.0, 1.0), (2, 2))
+        input_data_1d = np.arange(1.0, 5.0, 1.0)
+        if param_name in insert_some_nans:
+            input_data_2d[0, 1] = np.nan
+            input_data_2d[1, 0] = np.nan
+            input_data_1d[1] = np.nan
+        elif param_name in insert_all_nans:
+            input_data_2d = np.ones((2, 2)) * np.nan
+            input_data_1d = np.ones(4) * np.nan
+        input_data_2d *= magnitude
+        input_data_2d *= unit
+        input_data_1d *= magnitude
+        input_data_1d *= unit
+        input_data_0d = input_data_1d[3]
+        return input_data_0d, input_data_1d, input_data_2d
```

Comment:
This is neat, I've got two thoughts here.

Firstly, I was thinking that this would be a nice candidate for a pytest fixture (finally!), but then again... the neat thing about this is that it's generated dynamically, and fixtures - at least, I can't think of a way they could handle that.

Secondly, this could be for a future improvement, but: do you think we could also test 3d arrays here? Those will probably be a common case here.

Reply:
We could do 3d arrays, I suppose, I was thinking that 0d, 1d, N-d would be good enough, but it might be good to explicitly test 0, 1, 2, and 3 since those correspond to euclidean space and will likely be used.  We could even go 4 to get time in there.  I don't know... on one hand, testing each might be overkill, but sometimes overkill is underrated.